typed-racket-test/succeed/prefab.rkt

```diff
@@ -53,3 +53,45 @@
 ;; Test match (indirectly tests unsafe-struct-ref)
 (match (foo 'x) [(foo s) s])
 (match (foo* 'x) [(foo* s) s])
+
+
+
+(struct msg ([str : String] [maybe-str : (Option String)]) #:prefab)
+
+;; tests let bindings fields of prefab structs
+;; works as expected (requires path-type.rkt to be correct w.r.t. prefabs)
+(: test-let-binding (-> Any String))
+(define (test-let-binding x)
+  (cond
+    [(msg? x)
```

Comment:
This example doesn't seem sound to me. What if we just read an instance of MSG from a string with the wrong kind of fields?